client/dock.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017 Huawei Technologies Co., Ltd. All Rights Reserved.
+//
+//    Licensed under the Apache License, Version 2.0 (the "License"); you may
+//    not use this file except in compliance with the License. You may obtain
+//    a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//    License for the specific language governing permissions and limitations
+//    under the License.
+
+package client
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/opensds/opensds/pkg/model"
+)
+
+type DockMgr struct {
+	Receiver
+
+	Endpoint string
+	Opt      map[string]string
+	lock     sync.Mutex
+}
+
+func NewDockMgr(edp string) *DockMgr {
+	return &DockMgr{
+		Receiver: NewReceiver(),
+		Endpoint: edp,
+	}
+}
+
+func (d *DockMgr) GetDock(dckID string) (*model.DockSpec, error) {
+	var res model.DockSpec
+	url := d.Endpoint + "/api/v1alpha/docks/" + dckID
+
+	if err := d.Recv(request, url, "GET", d.Opt, &res); err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+
+	return &res, nil
+}
+
+func (d *DockMgr) ListDocks() ([]*model.DockSpec, error) {
+	var res []*model.DockSpec
+	url := d.Endpoint + "/api/v1alpha/docks"
+
+	if err := d.Recv(request, url, "GET", d.Opt, &res); err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func (d *DockMgr) ResetAndUpdateDockRequestContent(in interface{}) error {
```

Comment:
This method name  is too loog, it`s not good for reading, can you make a shorter one?

Reply:
Ok, I'll have a try, thanks jerry!